stability-tests/mempool-limits/transactions.go

```diff
@@ -0,0 +1,186 @@
+	"github.com/kaspanet/kaspad/stability-tests/common/mine"
+	"github.com/kaspanet/kaspad/util"
+	"strings"
+)
+
+const (
+	payAddress                       = "kaspasim:qr79e37hxdgkn4xjjmfxvqvayc5gsmsql2660d08u9ej9vnc8lzcywr265u64"
+	payAddressPrivateKey             = "0ec5d7308f65717f3f0c3e4d962d73056c1c255a16593b3989589281b51ad5bc"
+	fundingCoinbaseTransactionAmount = 1000
+	outputsPerTransaction            = 3
+	transactionFee                   = 1000
+	coinbaseMaturity                 = 100
+)
+
+var (
+	payAddressKeyPair     = decodePayAddressKeyPair()
+	payToPayAddressScript = buildPayToPayAddressScript()
+
+	fundingCoinbaseTransactions = make([]*externalapi.DomainTransaction, fundingCoinbaseTransactionAmount)
+)
+
+func generateFundingCoinbaseTransactions(rpcClient *rpcclient.RPCClient) {
+	// Generate one coinbase transaction for its side effect:
+	// the block containing it accepts the empty genesis coinbase
+	generateCoinbaseTransaction(rpcClient)
+
+	log.Infof("Generating funding coinbase transactions")
+	for i := 0; i < fundingCoinbaseTransactionAmount; i++ {
+		fundingCoinbaseTransactions[i] = generateCoinbaseTransaction(rpcClient)
+	}
+
+	log.Infof("Maturing funding coinbase transactions")
+	for i := 0; i < coinbaseMaturity; i++ {
+		generateCoinbaseTransaction(rpcClient)
+	}
+}
+
+func submitAnAmountOfTransactionsToTheMempool(
+	rpcClient *rpcclient.RPCClient, amountToSubmit int, ignoreOrphanRejects bool) {
+
+	log.Infof("Generating %d transactions", amountToSubmit)
+	transactions := make([]*externalapi.DomainTransaction, 0)
+	for len(transactions) < amountToSubmit {
+		var coinbaseTransaction *externalapi.DomainTransaction
+		coinbaseTransaction, fundingCoinbaseTransactions = fundingCoinbaseTransactions[0], fundingCoinbaseTransactions[1:]
+
+		unspentTransactions := []*externalapi.DomainTransaction{coinbaseTransaction}
+		for len(transactions) < amountToSubmit && len(unspentTransactions) > 0 {
+			var transactionToSpend *externalapi.DomainTransaction
+			transactionToSpend, unspentTransactions = unspentTransactions[0], unspentTransactions[1:]
+			spendingTransactions := generateTransactionsWithLotsOfOutputs(transactionToSpend)
+			transactions = append(transactions, spendingTransactions...)
+			unspentTransactions = append(unspentTransactions, spendingTransactions...)
+		}
+		log.Infof("Generated %d transactions", len(transactions))
+	}
+
+	transactions = transactions[:amountToSubmit]
+	log.Infof("Submitting %d transactions", len(transactions))
+
+	for i, transaction := range transactions {
+		rpcTransaction := appmessage.DomainTransactionToRPCTransaction(transaction)
+		_, err := rpcClient.SubmitTransaction(rpcTransaction)
+		if err != nil {
+			if ignoreOrphanRejects && strings.Contains(err.Error(), "orphan") {
+				continue
+			}
+			panic(err)
+		}
+		log.Infof("Submitted %d transactions", i+1)
+	}
+}
+
+func generateCoinbaseTransaction(rpcClient *rpcclient.RPCClient) *externalapi.DomainTransaction {
```

Comment:
I'd rename to `mineBlockAndGetCoinbaseTransaction` or something.
`generateCoinbaseTransaction` is only 10% of what this function actually does.